### PR TITLE
Do not need to strictly validate request params in Presign API.

### PR DESCRIPTION
**Pull request by harshavardhana**
This is done so that we can enable the existing API to work for
all API operations under S3.

### CODE REVIEW

File: api-presigned.go [hunk 1]
```diff
@@ -45,33 +35,18 @@ func (c Client) presignURL(method string, bucketName string, objectName string,
 	if err := s3utils.CheckValidBucketName(bucketName); err != nil {
 		return nil, err
 	}
-	if err := s3utils.CheckValidObjectName(objectName); err != nil {
-		return nil, err
-	}
```
Comment (poornas) [inline api-presigned.go]: why should object name validation be relaxed?

Comment (harshavardhana) [reply api-presigned.go]: To make it generic, need to move this check to presignedGetObject()

**Comment by harshavardhana** [comment]
This is ready for review again @poornas @krisis

File: api-presigned.go [hunk 2]
```diff
@@ -18,66 +18,42 @@ package minio
...
 func (c Client) presignURL(method string, bucketName string, objectName string, expires time.Duration, reqParams url.Values) (u *url.URL, err error) {
 	// Input validation.
 	if method == "" {
 		return nil, ErrInvalidArgument("method cannot be empty.")
 	}
-	if err := s3utils.CheckValidBucketName(bucketName); err != nil {
-		return nil, err
-	}
-	if err := s3utils.CheckValidObjectName(objectName); err != nil {
+	if err = s3utils.CheckValidBucketName(bucketName); err != nil {
```
Comment (krisis) [inline api-presigned.go]: Which http methods' presignURL doesn't require a valid object name?

Comment (harshavardhana) [reply api-presigned.go]: listObjects, headBucket, getBucketPolicy,getBucketLocation,.. there are more APIs.

Comment (krisis) [reply api-presigned.go]: Ok, makes sense.

**Comment by krisis** [comment]
@harshavardhana, is it possible to add a unit test case at least cover negative test cases like invalid bucket and object names?

**Comment by harshavardhana** [comment]
> @harshavardhana, is it possible to add a unit test case at least cover negative test cases like invalid bucket and object names?

Yes will do @krisis good idea.
